### ISSUE TITLE
Do not transform static JS assets

### ISSUE THREAD

**Issue by erke-soptim** [facebook/docusaurus#10460]
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#issues).

### Prerequisites

- [X] I'm using the latest version of Docusaurus.
- [X] I have tried the `npm run clear` or `yarn clear` command.
- [X] I have tried `rm -rf node_modules yarn.lock package-lock.json` and re-installing packages.
- [ ] I have tried creating a repro with https://new.docusaurus.io.
- [X] I have read the console error message carefully (if applicable).

### Description

Static assets ending in `.js` seem to be modified in a `docusaurus build`.

### Steps to reproduce

Create a new Docusaurus project.

```sh
npx create-docusaurus@latest my-website classic --typescript
cd my-website
```

Create a static asset containing some JavaScript.

```sh
echo "let a = 1; let b = 2;" > ./static/example.js
```

Build and serve the project.

```sh
npm run build
npm run serve
```

Open `localhost:3000/example.js` in your browser and observe that the content served by the browser does not look like the original file anymore, but like this.

```js
let a=1,b=2;
```

This does not happen when renaming the file to use a different extension, such as `.txt`.

### Expected behavior

I would expect Docusaurus not to modify the contents of static `.js` assets.

### Actual behavior

Docusaurus actually transformed the asset.

### Your environment

* Docusaurus version used: *3.5.2*
* Result of `uname -a`: *MSYS_NT-10.0-19045 NB01508 3.4.10-2e2ef940.x86_64 2024-07-09 21:35 UTC x86_64 Msys*
* Environment: *Chrome Version 128.0.6613.113 (Official Build) (64-bit)*

### Self-service

- [X] I'd be willing to fix this bug myself.

**Comment by Josh-Cena** [comment]
Static files should never be changed. They could be example code that's downloaded. I thought there's a similar issue but I can't find it now.

**Comment by Josh-Cena** [comment]
Ah, https://github.com/facebook/docusaurus/issues/10334 should be similar. Perhaps they are the same issue.

**Comment by soptim-erke** [comment]
> Static files should never be changed.

That is what I understood from [the docs](https://docusaurus.io/docs/static-assets#caveats) saying that *none of the files in the static folder will be post-processed, hashed, or minified*.

> They could be example code that's downloaded. I thought there's a similar issue but I can't find it now.

That is exactly what we are trying to achieve. We have a bunch of code snippets in the `static` folder that users can view and download. So, naturally, we would like to preserve how those are formatted.

**Comment by hrumhurum** [comment]
Perhaps related to https://github.com/facebook/docusaurus/issues/10334 _"Docusaurus 3.4.0+ does not respect PostCSS plugins when it minimizes CSS files from 'static' folder"_.

**Comment by allenscha** [comment]
As mentioned above, JS and CSS files in the static folder are getting minified since Docusaurus 3.4.

In our case, we put API reference files in the `static` folder, which include JS scripts and CSS styles generated by document generators such as Dokka, Jazzy, and the like. In one particular case, the minified assets are actually causing an issue where display of API reference is broken, which shows up properly in the build with Docusaurs 3.3.2 or lower.

In conclusion, I agree with @soptim-erke and @hrumhurum's comments on leaving the static resources intact.

**Comment by slorber** [comment]
fixed in [3.6.1](https://github.com/facebook/docusaurus/releases/tag/v3.6.1)
